src/Agent.Listener/Program.cs

```diff
@@ -102,22 +106,90 @@ public async static Task<int> MainAsync(IHostContext context, string[] args)
                     }
                 }
 
-                // Parse the command line args.
-                var command = new CommandSettings(context, args, new SystemEnvironment());
-                trace.Info("Arguments parsed");
+                Debugger.Launch();
 
-                // Up front validation, warn for unrecognized commandline args.
-                var unknownCommandlines = command.Validate();
-                if (unknownCommandlines.Count > 0)
+                // Accepted Commands
+                Type[] verbTypes = new Type[]
```

Comment:
So, personally, I would create a class or use the CommandArgs class and put all of this boiler plate stuff there. Perhaps in a static method that just returned the CommandSettings object. That way the implmentation is completely isolated.

Reply:
Now that I have seen how things are used and how the code is laid out. I would probably create a folder for "Commands" or "CommandLine" and then put all the new classes there. I would even create a separate file for each Command verb instead of putting them all in one file. But this is mostly personal style.